app/src/main/java/com/example/concordia_campus_guide/Activities/SearchActivity.java

```diff
@@ -1,23 +1,95 @@
 package com.example.concordia_campus_guide.Activities;
 
+import android.content.Intent;
+import android.location.Location;
 import android.os.Bundle;
+import android.text.Editable;
+import android.text.TextWatcher;
 import android.view.View;
+import android.widget.AdapterView;
+import android.widget.EditText;
 import android.widget.ImageButton;
+import android.widget.ListView;
 
 import androidx.appcompat.app.AppCompatActivity;
+import androidx.lifecycle.ViewModelProviders;
 
+import com.example.concordia_campus_guide.Adapters.PlaceToSearchResultAdapter;
+import com.example.concordia_campus_guide.Global.SelectingToFromState;
+import com.example.concordia_campus_guide.Models.Building;
+import com.example.concordia_campus_guide.Models.Floor;
+import com.example.concordia_campus_guide.Models.MyCurrentPlace;
+import com.example.concordia_campus_guide.Models.Place;
+import com.example.concordia_campus_guide.Models.RoomModel;
 import com.example.concordia_campus_guide.R;
 
 public class SearchActivity extends AppCompatActivity {
 
+    ListView searchResults;
+    EditText searchText;
+    SearchActivityViewModel mViewModel;
+
+    private PlaceToSearchResultAdapter adapter;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.search_activity);
 
+        //setting up activity
+        setContentView(R.layout.search_activity);
         setBackButtonOnClick();
+        searchResults = (ListView) findViewById(R.id.searchResults);
+        searchText = (EditText) findViewById(R.id.searchText);
+        mViewModel = ViewModelProviders.of(this).get(SearchActivityViewModel.class);
+
+        //android adapter for list view
+        adapter = new PlaceToSearchResultAdapter(this, R.layout.list_item_layout, mViewModel.getAllPlaces());
+        searchResults.setAdapter(adapter);
+
+        //search results filtering according to search text
+        searchText.addTextChangedListener(new TextWatcher(){
+            @Override
+            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2){}
+
+            @Override
+            public void onTextChanged(CharSequence charSequence, int i, int i1, int i2){
+                (SearchActivity.this).adapter.getFilter().filter(charSequence);
+            }
+            @Override
```

Comment:
the OnCreate method I feel that we can refactor it in a better way. For example, if you look in the LocationFragment file, we have a method called initComponent() which takes the responsibility of just initiating the components and then been called in the onCreate(). Furthermore,  you can also put the beforeTextChanged, onTextChanged and the aftertextChanged in a separate method n call it in create. What do you think?

Reply:
yep fixing it